azure-common/azure/profiles/__init__.py

```diff
@@ -0,0 +1,127 @@
+    *not* be considered as stable in the current implementation.
+
+    :param dict profile_dict: A profile dictionnary
+    :param str label: A label for pretty printing
+    """
+    def __init__(self, profile_dict, label=None):
+        self._profile_dict = profile_dict
+        self._label = label
+
+    @property
+    def label(self):
+        """The label associated to this profile definition.
+        """
+        return self._label
+
+    def __repr__(self):
+        return self._label if self._label else self._profile_dict.__repr__()
+
+    def get_profile_dict(self):
+        """Return the current profile dict.
+
+        This is internal information, and content should not be considered stable.
+        """
+        return self._profile_dict
+
+
+class DefaultProfile(object):
+    """Store a default profile.
+
+    :var ProfileDefinition profile: The default profile as class attribute
+    """
+    profile = None
+
+    def use(self, profile):
+        """Define a new default profile."""
+        if not isinstance(profile, KnownProfiles) and not isinstance(profile, ProfileDefinition):
+            raise ValueError("Can only set as default a ProfileDefinition or a KwownProfiles")
+        type(self).profile = profile
+
+    def definition(self):
+        return type(self).profile
+
+class KnownProfiles(Enum):
+    """This defines known Azure Profiles.
+
+    There is two meta-profiles:
+
+    - latest : will always use latest available api-version on each package
+    - default : mutable, will define profile automatically for all packages
+
+    If you change default, this changes all created packages on the fly to
+    this profile. This can be used to switch a complete set of API Version
+    without re-creating all clients.
+    """
+
+    # default - This is a meta-profile and point to another profile
+    default = DefaultProfile()
+    # latest - This is a meta-profile and does not contain definitions
+    latest = ProfileDefinition(None, "latest")
+    v2017_03_09_profile = ProfileDefinition(
+        {
+            "azure.mgmt.compute.ComputeManagementClient": {
+                None: "2016-03-30"
+            },
+            "azure.mgmt.network.NetworkManagementClient": {
+                None: "2015-06-15"
+            },
+            "azure.mgmt.storage.StorageManagementClient": {
+                None: "2016-01-01"
+            },
+            "azure.mgmt.resource.policy.PolicyClient": {
+                None: "2015-10-01-preview"
+            },
+            "azure.mgmt.resource.locks.ManagementLockClient": {
+                None: "2015-01-01"
+            },
+            "azure.mgmt.resource.links.ManagementLinkClient": {
+                None: "2016-09-01"
+            },
+            "azure.mgmt.resource.resources.ResourceManagementClient": {
+                None: "2016-02-01"
+            },
+            "azure.mgmt.resource.subscriptions.SubscriptionClient": {
+                None: "2016-06-01"
```

Comment:
What's the `None` key for?
I'm not familiar.

Reply:
This means default value. Using `None` allows to be sure we have no conflict, and there is only one value possible.